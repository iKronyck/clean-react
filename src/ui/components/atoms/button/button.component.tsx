import React from "react";
import { Button as MuiButton } from "@mui/material";

import { ButtonProps } from "./button.types";

export const Button = ({ label, ...props }: ButtonProps) => (
  <MuiButton {...props}>{label}</MuiButton>
);
