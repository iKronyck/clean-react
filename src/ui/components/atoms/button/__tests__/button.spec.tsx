import { render } from "@testing-library/react";

import { Button } from "../button.component";
import { ButtonProps } from "../button.types";

const makeSut = (props: ButtonProps) => render(<Button {...props} />);

describe("Button", () => {
  it("Should match snapshot", () => {
    const { container } = makeSut({ label: "label" });

    expect(container).toMatchSnapshot();
  });
});
