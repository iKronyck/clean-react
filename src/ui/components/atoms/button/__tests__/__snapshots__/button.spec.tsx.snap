// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root-KjFju gUleGd MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiButton-root-gygCtr esFZWn"
    tabindex="0"
    type="button"
  >
    label
    <span
      class="MuiTouchRipple-root-ieAhOC iowBRm MuiTouchRipple-root"
    />
  </button>
</div>
`;
